delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Acls.scala

```diff
@@ -0,0 +1,225 @@
+    * @param rev    the revision to fetch
+    */
+  final def fetchSelfAt(target: Target, rev: Long)(implicit caller: Caller): IO[RevisionNotFound, Option[AclResource]] =
+    fetchAt(target, rev).map(filterSelf)
+
+  /**
+    * Fetches the ACL for a ''target''. If ACL does not exist, return an empty [[Acl]]
+    *
+    * @param target the target location for the ACL
+    */
+  final def fetchAcl(target: Target): UIO[Acl] =
+    fetch(target).map(_.fold(Acl.empty)(_.value))
+
+  /**
+    * Fetches the ACL for a ''target''. If ACL does not exist, return an empty [[Acl]]
+    * The response only contains ACL with identities present in the provided ''caller''.
+    *
+   * @param target the target location for the ACL
+    */
+  final def fetchSelfAcl(target: Target)(implicit caller: Caller): UIO[Acl] =
+    fetchSelf(target).map(_.fold(Acl.empty)(_.value))
+
+  /**
+    * Fetches the [[AclTargets]] of the provided ''target'' location with some filtering options.
+    *
+    * @param target    the target location where the ACLs are going to be looked up
+    * @param ancestors flag to decide whether or not ancestor target locations should be included in the response
+    * @param self      flag to decide whether or not ancestor from other identities than the provided ones should be included in the response
+    * @param caller    the caller that contains the provided identities
+    */
+  def list(target: Target, ancestors: Boolean, self: Boolean)(implicit caller: Caller): UIO[AclTargets]
+
+  /**
+    * Overrides ''acl'' on a ''target''.
+    *
+    * @param target the target location for the ACL
+    * @param acl    the identity to permissions mapping to replace
+    * @param rev    the last known revision of the resource
+    */
+  def replace(target: Target, acl: Acl, rev: Long): IO[AclRejection, AclResource]
+
+  /**
+    * Appends ''acl'' on a ''target''.
+    *
+    * @param target the target location for the ACL
+    * @param acl    the identity to permissions mapping to append
+    * @param rev    the last known revision of the resource
+    */
+  def append(target: Target, acl: Acl, rev: Long): IO[AclRejection, AclResource]
+
+  /**
+    * Subtracts ''acl'' on a ''target''.
+    *
+    * @param target the target location for the ACL
+    * @param acl    the identity to permissions mapping to subtract
+    * @param rev    the last known revision of the resource
+    */
+  def subtract(target: Target, acl: Acl, rev: Long): IO[AclRejection, AclResource]
+
+  /**
+    * Delete all ''acl'' on a ''target''.
+    *
+    * @param target the target location for the ACL
+    * @param rev    the last known revision of the resource
+    */
+  def delete(target: Target, rev: Long): IO[AclRejection, AclResource]
+
+  private def filterSelf(resourceOpt: Option[AclResource])(implicit caller: Caller): Option[AclResource] =
+    resourceOpt.map(res => res.map(_.filter(caller.identities)))
+
+}
+
+object Acls {
```

Comment:
where are we handling the default permissions for `/`, when no acls are set?

Reply:
Currently it happens inside the `Acls`. But I can add it to the state machine